bin/Data/Scripts/15_Navigation.as

```diff
@@ -314,6 +314,8 @@ bool Raycast(float maxDistance, Vector3& hitPos, Drawable@& hitDrawable)
     if (!ui.cursor.visible || ui.GetElementAt(pos, true) !is null)
         return false;
 
+    pos = ui.ConvertUIToSystem(ui.cursorPosition);

```

Comment:
This is scene related stuff, just use input->mousePos

Reply:
Nope. According to my tests, `Input::GetMousePosition()` won't show the correct position if `MM_FREE ` is NOT used.

You could check `15_Natigation` and `39_CrowdNavigation`, the system mouse position remains in the center of the window while UI cursor is moving around, which leads to functional failure in our cases.

It seems to me that `UI::GetCursorPosition()` is origianlly designed to be the interface of get cursor position, just as the name shows, and scaled UI introduces some vagueness.

Reply:
Yes, it looks like you are right